src/Saritasa.Tools.Common/Utils/CollectionUtils.cs

```diff
@@ -640,5 +640,32 @@ public static void ApplyDiff<T>(ICollection<T> source, DiffResult<T> diff, Actio
                 return obj2;
             });
         }
+
+        /// <summary>
+        /// Iterate over a collection handling 2 items at a time, you always have the "previous" and "current" item.
+        /// </summary>
+        /// <typeparam name="T">Item source type.</typeparam>
+        /// <param name="source">Source collection.</param>
+        /// <returns>Enumeration in pairs.</returns>
+        /// <example>
+        /// Array [1, 2, 3, 4] will be transformed to [(1, 2), (2, 3), (3, 4)].
+        /// </example>
+        public static IEnumerable<(T, T)> Pairwise<T>(IEnumerable<T> source)
```

Comment:
1. Check for null.
2. Can we use tuples with names: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples#tuple-field-names ?

Reply:
Per discussion, tuple with names is not necessary